emu/src/memory/internal_memory.rs

```diff
@@ -209,7 +215,7 @@ impl InternalMemory {
 }
 
 impl IoDevice for InternalMemory {
-    type Address = u32;
+    type Address = usize;
     type Value = u8;
 
     fn read_at(&self, address: Self::Address) -> Self::Value {
```

Comment:
We should add the case when we read from the ROM mapped area, right? (The one starting at `0x08...`)

Reply:
I can add it :)

Reply:
Yup I think it makes sense to add it to this PR since without it we can't read instructions from the ROM

Reply:
Done